ignore missing hooks

[INFO] 2024/06/15 22:19:01 Unpackerr v0.13.1 Starting! PID: 1, UID: 0, GID: 0, Now: 2024-06-15 22:19:01 +0200 CEST
2024/06/15 22:19:01 [ERROR] cmdhook without a command configured; fix it

Origin user: colonelsandfurz.
Origin message: [Discord Server Link](https://discord.com/channels/184509568867368963/781625367391371274/1251632103133675562)
Reporting user: twitchcaptain
Reporting message: [Discord Server Link](https://discord.com/channels/184509568867368963/781625367391371274/1251632831571165336)

--- a/.github/workflows/codetests.yml
+++ b/.github/workflows/codetests.yml
@@ -39,7 +39,7 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v6
         with:
-          version: 'v1.55'
+          version: 'v1.59'
 
   golangci-linux:
     # description: "Runs golangci-lint on linux against linux and windows."
@@ -60,4 +60,4 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v6
         with:
-          version: 'v1.55'
+          version: 'v1.59'

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -2,21 +2,8 @@ linters:
   enable-all: true
   disable:
     # deprecated
-    - maligned
-    - scopelint
-    - interfacer
-    - golint
-    - exhaustivestruct
-    - nosnakecase
-    - structcheck
-    - deadcode
-    - varcheck
-    - ifshort
-    # unneeded (broken because of generics)
-    - rowserrcheck
-    - wastedassign
-    - sqlclosecheck
-    - contextcheck
+    - execinquery
+    - gomnd
     # unused
     - exhaustruct
     - exhaustive

--- a/pkg/ui/ui_other.go
+++ b/pkg/ui/ui_other.go
@@ -3,6 +3,7 @@
 package ui
 
 import (
+	"errors"
 	"fmt"
 	"io"
 	"os/exec"
@@ -27,7 +28,7 @@ func StartCmd(c string, v ...string) error {
 }
 
 // ErrUnsupported is just an error.
-var ErrUnsupported = fmt.Errorf("unsupported OS")
+var ErrUnsupported = errors.New("unsupported OS")
 
 // OpenCmd opens anything.
 func OpenCmd(_ ...string) error {

--- a/pkg/unpackerr/apps.go
+++ b/pkg/unpackerr/apps.go
@@ -1,6 +1,7 @@
 package unpackerr
 
 import (
+	"errors"
 	"fmt"
 	"strings"
 	"sync"
@@ -32,7 +33,7 @@ const (
 
 // Application validation errors.
 var (
-	ErrInvalidURL = fmt.Errorf("provided application URL is invalid")
+	ErrInvalidURL = errors.New("provided application URL is invalid")
 	ErrInvalidKey = fmt.Errorf("provided application API Key is invalid, must be %d characters", apiKeyLength)
 )
 
@@ -78,7 +79,7 @@ type workThread struct {
 
 func (u *Unpackerr) watchWorkThread() {
 	workers := u.Parallel
-	if workers > 4 { //nolint:gomnd // 4 == the four starr apps.
+	if workers > 4 { //nolint:mnd // 4 == the four starr apps.
 		workers = 4
 	}
 
@@ -122,16 +123,23 @@ func (u *Unpackerr) retrieveAppQueues() {
 // validateApps is broken-out into this file to make adding new apps easier.
 func (u *Unpackerr) validateApps() error {
 	for _, validate := range []func() error{
-		u.validateCmdhook,
 		u.validateLidarr,
 		u.validateRadarr,
 		u.validateReadarr,
 		u.validateSonarr,
 		u.validateWhisparr,
-		u.validateWebhook,
 	} {
 		if err := validate(); err != nil {
 			return err
+		}
+	}
+
+	for _, validate := range []func() error{
+		u.validateCmdhook,
+		u.validateWebhook,
+	} {
+		if err := validate(); err != nil {
+			u.Errorf("Config Warning: %v", err)
 		}
 	}
 

--- a/pkg/unpackerr/cmdhook.go
+++ b/pkg/unpackerr/cmdhook.go
@@ -63,6 +63,10 @@ func (u *Unpackerr) runCmdhookWithLog(hook *WebhookConfig, payload *WebhookPaylo
 }
 
 func (u *Unpackerr) runCmdhook(hook *WebhookConfig, payload *WebhookPayload) (*bytes.Buffer, error) {
+	if hook.Command == "" {
+		return nil, ErrCmdhookNoCmd
+	}
+
 	payload.Config = hook
 
 	env, err := cnfg.MarshalENV(payload, "UN")

--- a/pkg/unpackerr/cmdhook.go
+++ b/pkg/unpackerr/cmdhook.go
@@ -3,6 +3,7 @@ package unpackerr
 import (
 	"bytes"
 	"context"
+	"errors"
 	"fmt"
 	"os"
 	"os/exec"
@@ -15,7 +16,7 @@ import (
 
 // Errors produced by this file.
 var (
-	ErrCmdhookNoCmd = fmt.Errorf("cmdhook without a command configured; fix it")
+	ErrCmdhookNoCmd = errors.New("cmdhook without a command configured; fix it")
 )
 
 func (u *Unpackerr) validateCmdhook() error {
@@ -47,13 +48,12 @@ func (u *Unpackerr) runCmdhookWithLog(hook *WebhookConfig, payload *WebhookPaylo
 
 	hook.Lock() // we only lock for the integer increments.
 	defer hook.Unlock()
-
-	hook.posts++
+	hook.posts++ //nolint:wsl
 
 	switch {
 	case err != nil:
 		u.Errorf("Command Hook (%s) %s: %v: %s", payload.Event, hook.Name, err, out.String())
-		hook.fails++
+		hook.fails++ //nolint:wsl
 	case hook.Silent || out == nil:
 		u.Printf("[Cmdhook] Queue: %d/%d. Ran command %s", len(u.hookChan), cap(u.hookChan), hook.Name)
 	default:
@@ -122,7 +122,7 @@ func (u *Unpackerr) logCmdhook() {
 		pfx = " => Command Hook Config: 1 cmd"
 	} else {
 		u.Printf(" => Command Hook Configs: %d commands", len(u.Cmdhook))
-		pfx = " =>    Command"
+		pfx = " =>    Command" //nolint:wsl
 	}
 
 	for _, f := range u.Cmdhook {

--- a/pkg/unpackerr/folder.go
+++ b/pkg/unpackerr/folder.go
@@ -84,6 +84,7 @@ func (u *Unpackerr) logFolders() {
 			if epath = ""; folder.ExtractPath != "" {
 				epath = ", extract to: " + folder.ExtractPath
 			}
+
 			u.Printf(" =>    Path: %s%s (delete after:%v, delete orig:%v, log file: %v, move back:%v, isos:%v)",
 				folder.Path, epath, folder.DeleteAfter, folder.DeleteOrig, !folder.DisableLog, folder.MoveBack, folder.ExtractISOs)
 		}

--- a/pkg/unpackerr/webhook.go
+++ b/pkg/unpackerr/webhook.go
@@ -175,6 +175,10 @@ func (u *Unpackerr) sendWebhookWithLog(hook *WebhookConfig, payload *WebhookPayl
 
 // Send marshals an interface{} into json and POSTs it to a URL.
 func (w *WebhookConfig) Send(body io.Reader) ([]byte, error) {
+	if w.URL == "" {
+		return nil, ErrWebhookNoURL
+	}
+
 	w.Lock()
 	defer w.Unlock()
 

--- a/pkg/unpackerr/webhook.go
+++ b/pkg/unpackerr/webhook.go
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"context"
 	"crypto/tls"
+	"errors"
 	"fmt"
 	"io"
 	"net/http"
@@ -48,8 +49,8 @@ type hookQueueItem struct {
 
 // Errors produced by this file.
 var (
-	ErrInvalidStatus = fmt.Errorf("invalid HTTP status reply")
-	ErrWebhookNoURL  = fmt.Errorf("webhook without a URL configured; fix it")
+	ErrInvalidStatus = errors.New("invalid HTTP status reply")
+	ErrWebhookNoURL  = errors.New("webhook without a URL configured; fix it")
 )
 
 // ExtractStatuses allows us to create a custom environment variable unmarshaller.
@@ -176,6 +177,7 @@ func (u *Unpackerr) sendWebhookWithLog(hook *WebhookConfig, payload *WebhookPayl
 func (w *WebhookConfig) Send(body io.Reader) ([]byte, error) {
 	w.Lock()
 	defer w.Unlock()
+
 	w.posts++
 
 	ctx, cancel := context.WithTimeout(context.Background(), w.Timeout.Duration+time.Second)
@@ -195,7 +197,7 @@ func (w *WebhookConfig) send(ctx context.Context, body io.Reader) ([]byte, error
 		return nil, fmt.Errorf("creating request: %w", err)
 	}
 
-	req.Header.Set("content-type", w.CType)
+	req.Header.Set("Content-Type", w.CType)
 
 	res, err := w.client.Do(req)
 	if err != nil {
@@ -266,24 +268,24 @@ func (u *Unpackerr) logWebhook() {
 		pfx = " => Webhook Config: 1 URL"
 	} else {
 		u.Printf(" => Webhook Configs: %d URLs", len(u.Webhook))
-		pfx = " =>    URL"
+		pfx = " =>    URL" //nolint:wsl
 	}
 
 	for _, f := range u.Webhook {
 		if ex = ""; f.TmplPath != "" {
-			ex = fmt.Sprintf(", template: %s, content_type: %s", f.TmplPath, f.CType)
+			ex = ", template: " + f.TmplPath + ", content_type: " + f.CType
 		}
 
 		if f.Channel != "" {
-			ex += fmt.Sprintf(", channel: %s", f.Channel)
+			ex += ", channel: " + f.Channel
 		}
 
 		if f.Nickname != "" {
-			ex += fmt.Sprintf(", nickname: %s", f.Nickname)
+			ex += ", nickname: " + f.Nickname
 		}
 
 		if len(f.Exclude) > 0 {
-			ex += fmt.Sprintf(", exclude: %q", strings.Join(f.Exclude, "; "))
+			ex += ", exclude: \"" + strings.Join(f.Exclude, "; ") + `"`
 		}
 
 		u.Printf("%s: %s, timeout: %v, ignore ssl: %v, silent: %v%s, events: %q",

